result.go

```diff
@@ -142,22 +143,24 @@ func (r *result) setLabels(t *testing.T) {
 
 func (r *result) writeResultsFile() error {
 	resultsPathEnv := os.Getenv(resultsPathEnvKey)
+	once.Do(copyEnvFileIfExists)
```

Comment:
please move this to line 162 (before we create write the file), so we can remove the code creating the folder from copyEnvFileIfExists().
It would be a good time to put the code creating the folder into a once.Do().